cmd/clusterctl/client/tree/annotations.go

```diff
@@ -118,7 +118,19 @@ func IsShowConditionsObject(obj client.Object) bool {
 	return false
 }
 
-func getAnnotation(obj client.Object, annotation string) (string, bool) {
+// HasTruthyAnnotation evaluates truthiness permissively, only return false if
+// 1) the annotation isn't present at all
+// 2) the annotation is explicitly set to "false".
+// All other value permutations are ignored and considered valid.
+// tl;dr We're looking for boolean-type annotations but respecting that some folks might want to explicitly indicate "false".
+func HasTruthyAnnotation(obj client.Object, annotation string) bool {
```

Comment:
This probably belongs better in `util/annotations/helpers.go` since we're not using it in clusterctl (unsure why clusterctl has its own annotation helpers)

Reply:
Also, might be good to add a `func ReplicasManagedByExternalAutoscaler(o metav1.Object) bool {`  helper function in there, similar to what we have for other annotations already (and possibly reuse this definition of "hasTruthyAnnotation" across all annotations in CAPI, although that would be out of scope for this PR, but a great `help wanted` issue).

Reply:
done, I'm not sure using this "truthy" definition across capi isn't a bit controversial...

Reply:
agreed, that's why starting with an issue makes more sense, not saying we should do it in this PR (but I do think we want that change to happen, inconsistency is always worse IMO...)

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/7434